rust/datafusion/src/logicalplan.rs

```diff
@@ -178,42 +174,36 @@ pub enum Expr {
 }
 
 impl Expr {
-    pub fn get_type(&self, schema: &Schema) -> DataType {
+    pub fn get_type(&self, schema: &Schema) -> Result<DataType> {
```

Comment:
Every `Expr` should have a `DataType` associated with it, so it feels weird to return a `Result`.

Reply:
Again this is a case of some things just not being implemented yet. Maybe I should revert this one to still panic as well.